scripts/packages/VSCodeServer/src/progress.jl

```diff
@@ -0,0 +1,27 @@
+struct VSCodeLogger <: Logging.AbstractLogger end
+
+function Logging.handle_message(j::VSCodeLogger, level, message, _module,
+                                group, id, file, line; kwargs...)
+    progress = ProgressLogging.asprogress(level, message, _module, group, id, file, line; kwargs...)
+    if progress !== nothing
+        JSONRPC.send_notification(conn_endpoint[], "repl/updateProgress", progress)
+        return nothing
+    else
+        previous_logger = Logging.global_logger()
+        if (Base.invokelatest(Logging.min_enabled_level, previous_logger) <= Logging.LogLevel(level) ||
+            Base.CoreLogging.env_override_minlevel(group, _module)) &&
+            Base.invokelatest(Logging.shouldlog, previous_logger, level, _module, group, id)
+            Logging.handle_message(previous_logger, level, message, _module,
+                            group, id, file, line; kwargs...)
+        end
+    end
```

Comment:
I don't understand that whole `else` block, could you add some comments that just give a rough explanation of what it is doing? :)